File: cpp_src/msg_self/src/pub.cpp
#include <ros/ros.h>
#include <msg_self/Student.h>

int main(int argc, char **argv) {
    ros::init(argc, argv, "msg_pub");

    ros::NodeHandle nh;
    ros::Publisher msg_pub = nh.advertise<msg_self::Student>("student", 1000);

    ros::Rate loop_rate(10);
    while (ros::ok()) {
        msg_self::Student msg;
        msg.name = "jieshoudaxue";
        msg.age = 30;
        ROS_INFO("name = %s, age = %u", msg.name.c_str(), msg.age);

        msg_pub.publish(msg);

        ros::spinOnce();
        loop_rate.sleep();
    }

    return 0;
}

File: pyhon_src/msg_self/scripts/pub.py
#! /usr/bin/env python3

import rospy
from msg_self.msg import Student

def main():
    rospy.init_node("msg_pub")

    pub = rospy.Publisher("student", Student, queue_size=10)

    rate = rospy.Rate(10)

    while not rospy.is_shutdown():
        s1 = Student()
        s1.name = "jieshoudaxue"
        s1.age = 30

        pub.publish(s1)

        rospy.loginfo("send s1, name = %s, age = %d" %(s1.name, s1.age))
        rate.sleep()



if __name__ == "__main__":
    main()